src/core/unit.c

```diff
@@ -1575,6 +1605,12 @@ int unit_start(Unit *u) {
         if (!unit_supported(u))
                 return -EOPNOTSUPP;
 
+        /* Let's make sure that the deps really are in order before we start this. Normally the job engine should have
+         * taken care of this already, but let's check this here again. After all, the dependencies might have changed
+         * due to a reload or because the user forgot to place ordering dependencies in conjunction with BindsTo=. */
+        if (!unit_verify_deps(u))
+                return -ENOLINK;
```

Comment:
Indeed in this context it's not clear at all. You mostly want to say "stop b if a goes away", but don't care about the ordering, and with a normal `Requires=` it is *not* a requirement to add `After=` as well. It still makes sense in the context of `.device` units (although this is mostly for an internal/technical reason, in the sense that systemd needs to "import" udev devices), but not here indeed.